### PR TITLE
Add header navigation

### LINKED ISSUE #128: Implement new header for Next.js product list pages
## Problem

We've decided to build a new, simplified header for the collections page, rather than refactor our existing header code to work in both places. There will still need to be shared code eventually, but for the sake of an MVP we want to save the more complicated bits for later.

We'll build it from scratch with Chakra, TypeScript, Next links/images, etc. and pull in the logged-in user's info via the `/api/2.0/user` endpoint just like we do with the Shopify frontend currently.

## Do This

For MVP, the header should include:
- [x] iFixit Logo
- [x] Header categories and associated dropdown items
- [x] Search
      - Searches the US store, just like the existing product list page
- [x] Login / Join Buttons
      - Use `/login` and `/join` links instead of opening a modal
- [x] User Menu (for logged in users)
      - Do not include in menu:
          - Count for `Notifications` option 
          - `/Manage` link, or associated logic
- [x] Fix CLS issues by changing unauthenticated UI

## These have been moved after V1
- [ ] #150
- [ ] #151
- [ ] #152
- [ ] #153

## Testing notes

- links go to the correct URL
- all dropdowns work correctly

CC: @dhmacs

### PR DESCRIPTION
**Pull request by dhmacs**
The PR introduces a react based header bar (not fully featured yet, some parts like the cart and the workbench have been postponed).

The menu data is fetched from Strapi, so menu links can be added, removed or changed. Right now the UI supports a main menu with 3 submenus. In the future it could be extended to support other combination if needed (e.g. link with image).

The header doesn't reflect exactly the current iFixit header, since it introduces a few changes (e.g. "login" / "join" links have been removed in favor of a default user avatar that let us avoid CLS, since auth data is fetched client side).

## QA

1. Visit the Vercel PR preview
2. Visit parts product list page and copy the url
3. Verify that #128 requirements under "Do this" are met

closes #128

### CODE REVIEW

**Comment by ardelato** [comment]
QA 👍 


----

deploy_block ☁️  

   <details>
	<summary>1. What should be the expected behavior for login menu? </summary>
   
   
  Should the avatar change after logging into Cominor? 

  https://user-images.githubusercontent.com/22064420/145689302-1558586c-45b3-4744-86b2-f4b77611e326.mp4

   </details>

   <details>
	<summary>2. On mobile, the Avatar icon is not aligned vertically </summary>
   
     
  Mobile Misaligned avatar ico@2x

   </details>


  <details>
	<summary>3. On mobile, should the language and store option also be included in the header? </summary>
   

   |iFixit	| This Branch	|
   |:---:|:---:|
   |iFixit Mobile options@2x|React mobile options@2x|
   
   </details>


## Steps Taken:
1. Visited the Vercel Preview
2. Confirmed the header bar is now updated
3. Clicked on the iFixit logo and it took me back to the home page 👍 
Header bar there@2x

4. Hovered over the different menu bar options and confirmed that the dropdown options work on hover and are the same as on ifixit.com
Dropdown menus work@2x

5. I confirmed that the search functionality redirects you to iFixit as expected -- https://github.com/iFixit/react-commerce/issues/128#issuecomment-977196941
6. I then clicked on the User menu as was redirected to www.cominor.com/login, but after logging into Cominor I went back to the Vercel page and did not notice any differences in the User Menu icon? 
7. I then tested across different devices

**Comment by dhmacs** [comment]
@ardelato 

> 1. What should be the expected behavior for login menu?

Sorry, I should have mentioned here the slack conversation. I'm waiting on an answer to this, since it seems that there are CORS issue when we deploy on Vercel

Screenshot 2021-12-13 at 09 36 04


> 2. On mobile, the Avatar icon is not aligned vertically

I'll fix it 👍 


> 3. On mobile, should the language and store option also be included in the header?

I think we're doing the same we're doing for desktop, that is they will be in the footer. Anyway if we want to change our mind later we can always open another PR, this one already introduces quite a few changes

File: frontend/components/Layout/Navigation.tsx [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+      return (
+         <Box ref={ref} as="nav" h="full" ml="6" {...boxProps}>
+            <MenuBar h="full" position="relative">
+               {menu.items.map((item, index) => {
+                  switch (item.type) {
+                     case 'submenu': {
+                        return (
+                           <MenuItem
+                              key={index}
+                              transition="background-color 300ms"
```
Comment (masonmcelvain) [inline frontend/components/Layout/Navigation.tsx:46]: Not sure if this was intentional, but this transition could use `ANIMATION_DURATION` from above

File: frontend/components/Layout/Navigation.tsx [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+                                                               bg:
+                                                                  'trueGray.700',
+                                                               position:
+                                                                  'absolute',
+                                                               h: '100%',
+                                                               w: '4px',
+                                                               borderRadius:
+                                                                  'full',
+                                                               transition:
+                                                                  'background-color 300ms',
```
Comment (masonmcelvain) [inline frontend/components/Layout/Navigation.tsx:273]: `ANIMATION_DURATION`

File: frontend/components/Layout/Navigation.tsx [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+         mr="1"
+         _hover={{
+            bg: 'trueGray.800',
+         }}
+         _active={{
+            bg: 'trueGray.800',
+         }}
+         sx={{
+            '& path': {
+               transition: 'transform 300ms',
```
Comment (masonmcelvain) [inline frontend/components/Layout/Navigation.tsx:325]: `ANIMATION_DURATION`

File: frontend/components/Layout/SearchBar.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import * as React from 'react';
+import { RiSearchLine } from 'react-icons/ri';
+
+export const SearchForm = forwardRef<FlexProps, 'form'>((props, ref) => {
+   return (
+      <Flex
+         ref={ref}
+         as="form"
+         method="GET"
+         action="https://www.ifixit.com/Search"
```
Comment (masonmcelvain) [inline frontend/components/Layout/SearchBar.tsx:20]: Should we use `NEXT_PUBLIC_IFIXIT_ORIGIN` here?

File: frontend/components/Layout/User.tsx [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+                     >
+                        Log Out
+                     </MenuItem>
+                  </MenuGroup>
+               </MenuList>
+            </Menu>
+         ) : (
+            <Link borderRadius="full" href={`${IFIXIT_ORIGIN}/login`}>
+               <NoUserIcon
+                  transition="color 300ms"
```
Comment (masonmcelvain) [inline frontend/components/Layout/User.tsx:87]: If we're trying to use 300ms transition duration across components, maybe we should add it to the chakra theme?

File: frontend/lib/api/layout/index.ts [hunk 6]
```diff
@@ -74,6 +79,9 @@ function getMenu(rawMenu: RawMenu): Menu {
                      type: 'link',
                      name: item.name,
                      url: item.url,
+                     descriptionHtml: item.description
+                        ? snarkdown(item.description)
```
Comment (masonmcelvain) [inline frontend/lib/api/layout/index.ts:83]: cool!

**Comment by masonmcelvain** [comment]
CR ✌🏻 Looks good after a quick skim. Just a couple comments about using constants, don't need to be addressed in this PR

**Comment by dhmacs** [comment]
@ardelato @masonmcelvain 

Should be good to go now 👍

**Comment by ardelato** [comment]
QA 👍 


deploy_block 🎨  on UX issue

   <details>
	<summary> It seems there is a hidden <code>div</code> on mobile that is being used to create artificial spacing. </summary>
   
   
   <details>
	<summary> Screenshots</summary>

    Unneeded Div

    Flex
   
   </details>



Instead, we should remove it and declare the justification for the flex items:

   <details>
	<summary>Screenshots </summary>
   
   Justify Space between
   
   </details>


By changing the justification like this, the `iFixit` home link is now not too close to the `hamburger` menu. On mobile, the `hamburger` menu is pretty small and relatively close to the `iFixit` home link which if some has bigger fingers or a smaller phone, then there is a high possibility of mis-tapping the `hamburger` menu.

Overall though, the header menu on mobile should probably be revisited for better UX/UI design. 

   </details>

----

💭.   Also, will the login issue be resolved in another pull?
> The behaviour for the authenticated user is that you should see the user avatar. You're not seeing it because there are CORS issues between cominor and Vercel, I've asked for help in Slack:




## Steps Taken:
1. Repeated steps from https://github.com/iFixit/react-commerce/pull/154#issuecomment-991763247

**Comment by masonmcelvain** [comment]
CR ✌🏻

**Comment by dhmacs** [comment]
@ardelato 

>   It seems there is a hidden <code>div</code> on mobile that is being used to create artificial spacing. 

The header navigation definitely needs some redesign, but I think it should be pushed out of the scope of this PR. For now the goal is to have a functional nav bar with some of the features of the one that we have on ifixit.com (mobile version navbar of ifixit.com looks the same)

> Also, will the login issue be resolved in another pull?

It should be fixed now, is it not working for you?

**Comment by GrantMatejka** [comment]
CR 🌽

**Comment by ardelato** [comment]
> The header navigation definitely needs some redesign, but I think it should be pushed out of the scope of this PR. For now the goal is to have a functional nav bar with some of the features of the one that we have on ifixit.com (mobile version navbar of ifixit.com looks the same)

Roger that!

---

> It should be fixed now, is it not working for you?

Well I am not sure what to expect? I do see the request sent to retrieve the user information and it retrieves the user avatar. However, I do not see the avatar being used?


https://user-images.githubusercontent.com/22064420/147498669-04cb670f-d3cd-40d4-afcc-c0bed1aa06d6.mp4

**Comment by dhmacs** [comment]
@ardelato 

> @dhmacs, well I am not sure what to expect? I do see the request sent to retrieve the user information and it retrieves the user avatar. However, I do not see the avatar being used?

I think I've spotted the issue. I thought that `unique_username` was required, but apparently it's not? The edit profile page won't let me save a null / empty username. How come that yours is empty? Is it expected for a user not to have a username?

image

**Comment by ardelato** [comment]
@dhmacs nice catch! Adding a unique username resolved the issue 
User mentioned


The reason this user did not have a unique username is because it was not created through the normal `Sign Up` method but instead through a Dev tool that does not assign a unique user name 

Create Test User

un_deploy_block 👍

**Pull request merged**